concent_api/core/tests/utils.py

```diff
@@ -204,6 +209,50 @@ def _test_database_objects(
             self.assertEqual(ReceiveStatus.objects.last().delivered,        receive_delivered_status)
             self.assertEqual(ReceiveStatus.objects.last().message.task_id,  task_id)
 
+    def _test_subtask_state(
+        self,
+        task_id,
+        subtask_id,
+        subtask_state,
+        provider_key,
+        requestor_key,
+        next_deadline                   = None,
+        has_task_to_compute             = None,
+        has_report_computed_task        = None,
+        has_ack_report_computed_task    = None,
+        has_reject_report_computed_task = None,
+        has_subtask_results_accepted    = None,
+        has_subtask_results_rejected    = None,
+    ):
+        self.assertTrue(StoredMessage.objects.filter(subtask_id = subtask_id).exists())
+        subtask = Subtask.objects.get(subtask_id = subtask_id)
+        self.assertEqual(subtask.task_id,              task_id)
+        self.assertEqual(subtask.subtask_id,           subtask_id)
+        self.assertEqual(subtask.state,                subtask_state)
+        self.assertEqual(subtask.provider.public_key,  provider_key)
+        self.assertEqual(subtask.requestor.public_key, requestor_key)
+
+        self.assertTrue(Client.objects.filter(public_key = provider_key).exists())
+        self.assertTrue(Client.objects.filter(public_key = requestor_key).exists())
+
+        subtask_deadline = None
+        if subtask_state in Subtask.SUBTASK_ACTIVE_STATES:
+            subtask_deadline = subtask.next_deadline.timestamp()
+        self.assertEqual(subtask_deadline, next_deadline)
+
+        if has_task_to_compute:
```

Comment:
I think we should check if each message `is None` if `has_` parameters is `False`.